### PR TITLE
LPD-26931

### PR DESCRIPTION
**Pull request by daiazius**

### CODE REVIEW

**Comment by daiazius** [comment]
ci:test:sf

**Comment by ChrisKian** [comment]
Hey @daiazius,
Do we need to include an upgrade process to create the new SAP entry?  Testing locally, if I add your changes and deploy the module(s), the logic is there, but not the SAP entry.  Or, is there some step I am missing?

**Comment by ChrisKian** [comment]
Also, if I start with a clean bundle, the entry is created properly, but step 5 on LPD-26931 does not work:

> 5. Place the fragment in the Home page
>   a. Note that the page renders without any issues

I receive "Error:Error: An unexpected error occurred" instead of the fragment being added properly.

**Comment by daiazius** [comment]
> Also, if I start with a clean bundle, the entry is created properly, but step 5 on LPD-26931 does not work:
> 
> > 5. Place the fragment in the Home page
> >    a. Note that the page renders without any issues
> 
> I receive "Error:Error: An unexpected error occurred" instead of the fragment being added properly.

Hey @ChrisKian, I tested here and this indeed happens, I'm not sure why though. I'll update the ticket to reflect that.

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:sf

**Comment by daiazius** [comment]
ci:test:relevant

**Review by stian-sigvartsen — CHANGES_REQUESTED** [review]
The design looks good @daiazius !
I tested that when a SYSTEM_TEMPLATE_DEFAULT SAP entry exists it is activatd for the REST Client requests. Two things:

1. I wasn't able to get the SAP entry to be automatically created. Looking at invocations to `SAPEntryLocalServiceImpl#checkSystemSAPEntries` I found it called from `SAPServiceVerifyProcess` , but this did not automatically trigger upon deployment. Is it working differently for you?
2. We need to add tests

**Comment by daiazius** [comment]
Hi @stian-sigvartsen, thanks for reviewing!

About 1, I was also having trouble for the new entry being created automatically, from the code it should be creating when the module is deployed. I managed to get it created by cleaning my DB the first time I tried, but I don't think it should be this way, I need to investigate what triggers `SAPServiceVerifyProcess` or `VerifyProcess` in general.

About 2, I thought so, I'll start working on it.

**Comment by stian-sigvartsen** [comment]
After speaking with @ChrisKian , I think the SAP entry gets automatically added as part of the virtual instance creation. So it will also work with a clean DB also. If we continue with that current approach then we will need to add an upgrade process I think. `VerifyProcess` might also do it, but I think they need to be manually triggered by an admin.

**Comment by ChrisKian** [comment]
It looks like we will create the SAP entry whenever 1) [we (re)create the default password policy](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/model/listener/PasswordPolicyModelListener.java#L38-L39) or 2) [run the SAPServiceVerifyProcess](https://github.com/liferay/liferay-portal/blob/master/modules/apps/portal-security/portal-security-service-access-policy-service/src/main/java/com/liferay/portal/security/service/access/policy/internal/verify/SAPServiceVerifyProcess.java#L30).

I'm wondering if there is a way we can leverage the verify process?  Otherwise, and upgrade process will be needed.

**Comment by ChrisKian** [comment]
UpgradeProcess PR: https://github.com/liferay-appsec/liferay-portal/pull/1808

File: modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java [hunk 1]
```diff
@@ -51,6 +53,8 @@ public RESTClientHttpRequest(
 		HttpServletRequest httpServletRequest) {
 
 		_attributes = HashMapBuilder.<String, Object>put(
+			SAPWebKeys.TEMPLATE_DIPSATCH, true
```
Comment (stian-sigvartsen) [inline modules/apps/portal-vulcan/portal-vulcan-impl/src/main/java/com/liferay/portal/vulcan/internal/template/servlet/RESTClientHttpRequest.java:56]: `TEMPLATE_DISPATCH` not `TEMPLATE_DIPSATCH`.

**Comment by ChrisKian** [comment]
Resent here: https://github.com/liferay-appsec/liferay-portal/pull/1815

**Pull request closed without merging**